utils/forest_utils/src/db/mod.rs

```diff
@@ -68,3 +72,40 @@ pub trait BlockstoreExt: Blockstore {
 }
 
 impl<T: fvm_ipld_blockstore::Blockstore> BlockstoreExt for T {}
+
+/// Extension methods for buffered write with manageable limit of RAM usage
+#[async_trait]
+pub trait BlockstoreBufferedWriteExt: Blockstore + Sized {
+    async fn buffered_write(
+        &self,
+        rx: flume::Receiver<(Cid, Vec<u8>)>,
+        buffer_capacity_bytes: usize,
+    ) -> anyhow::Result<()> {
+        let start = Utc::now();
+        let mut total_bytes = 0;
+        let mut total_entries = 0;
+        let mut estimated_buffer_bytes = 0;
+        let mut buffer = vec![];
+        while let Ok((key, value)) = rx.recv_async().await {
+            // Key is stored in 32 bytes in paritydb
+            estimated_buffer_bytes += 32 + value.len();
+            total_bytes += 32 + value.len();
+            total_entries += 1;
+            buffer.push((key, value));
+            if estimated_buffer_bytes >= buffer_capacity_bytes {
+                self.put_many_keyed(std::mem::take(&mut buffer))?;
+                estimated_buffer_bytes = 0;
+            }
+        }
+        self.put_many_keyed(buffer)?;
+        info!(
```

Comment:
Is this instead a debug? Also, I'm not sure calling timers by default makes sense in a performance-related method.

Reply:
If it's helpful, perhaps let's have a proper Prometheus metric? Like, average speed or something like this.

Reply:
In this case, I think log of individual runs is more helpful than aggregated metrics to understand details and nature of the database and for troubleshooting.  How about keeping it as it is, given it only emits 2-3 lines every 1.5-2 days?

Reply:
Ok